File: src/javafile/Reserva.java
/**
 * @author álvaro Rodríguez
 * @version 0.0
 * @see <a href = "https://github.com/alvrichh" /> GitHub </a>
 */

package javafile;

import java.util.Date;

/**
 * Clase que representa una reserva.
 */
public class Reserva {
    /**
     * Usuario que realiza la reserva.
     */
    private String usuario;

    /**
     * Libro reservado.
     */
    private Libro libro;

    /**
     * Fecha de reserva.
     */
    private Date fechaReserva;

    /**
     * Fecha de devolucion.
     */
    private Date fechaDevolucion;

    /**
     * Constructor de la clase Reserva.
     * 
     * @param usuario         El usuario que realiza la reserva.
     * @param libro           El libro reservado.
     * @param fechaReserva    La fecha de reserva.
     * @param fechaDevolucion La fecha de devolución.
     */
    public Reserva(String usuario, Libro libro, Date fechaReserva, Date fechaDevolucion) {
        setUsuario(usuario);
        setLibro(libro);
        setFechaReserva(fechaReserva);
        setFechaDevolucion(fechaDevolucion);
    }

    /**
     * Obtiene el usuario que realizo la reserva.
     *
     * @return El usuario que realizo la reserva.
     */
    public String getUsuario() {
        return usuario;
    }

    /**
     * Establece el usuario que realizo la reserva.
     *
     * @param usuario El nuevo usuario que realizo la reserva.
     */
    public void setUsuario(String usuario) {
        this.usuario = usuario;
    }

    /**
     * Obtiene el libro reservado.
     *
     * @return El libro reservado.
     */
    public Libro getLibro() {
        return libro;
    }

    /**
     * Establece el libro reservado.
     *
     * @param libro El nuevo libro reservado.
     */
    public void setLibro(Libro libro) {
        this.libro = libro;
    }

    /**
     * Obtiene la fecha de reserva.
     *
     * @return La fecha de reserva.
     */
    public Date getFechaReserva() {
        return fechaReserva;
    }

    /**
     * Establece la fecha de reserva.
     *
     * @param fechaReserva La nueva fecha de reserva.
     */
    public void setFechaReserva(Date fechaReserva) {
        this.fechaReserva = fechaReserva;
    }

    /**
     * Obtiene la fecha de devolucion.
     *
     * @return La fecha de devolucion.
     */
    public Date getFechaDevolucion() {
        return fechaDevolucion;
    }

    /**
     * Establece la fecha de devolucion.
     *
     * @param fechaDevolucion La nueva fecha de devolucion.
     */
    public void setFechaDevolucion(Date fechaDevolucion) {
        this.fechaDevolucion = fechaDevolucion;
    }
}

File: src/js/Reserva.js
/**
 * Clase que representa una reserva.
 * @class
 */
class Reserva {
    /**
     * Constructor de la clase Reserva.
     * @constructor
     * @param {string} usuario - Usuario que realiza la reserva.
     * @param {Libro} libro - Libro reservado.
     * @param {Date} fechaReserva - Fecha de reserva.
     * @param {Date} fechaDevolucion - Fecha de devolución.
     */
    constructor(usuario, libro, fechaReserva, fechaDevolucion) {
        this.usuario = usuario;
        this.libro = libro;
        this.fechaReserva = fechaReserva;
        this.fechaDevolucion = fechaDevolucion;
    }

    /**
     * Obtiene el usuario que realizó la reserva.
     * @returns {string} El usuario que realizó la reserva.
     */
    getUsuario() {
        return this.usuario;
    }

    /**
     * Establece el usuario que realizó la reserva.
     * @param {string} usuario - El nuevo usuario que realizó la reserva.
     */
    setUsuario(usuario) {
        this.usuario = usuario;
    }

    /**
     * Obtiene el libro reservado.
     * @returns {Libro} El libro reservado.
     */
    getLibro() {
        return this.libro;
    }

    /**
     * Establece el libro reservado.
     * @param {Libro} libro - El nuevo libro reservado.
     */
    setLibro(libro) {
        this.libro = libro;
    }

    /**
     * Obtiene la fecha de reserva.
     * @returns {Date} La fecha de reserva.
     */
    getFechaReserva() {
        return this.fechaReserva;
    }

    /**
     * Establece la fecha de reserva.
     * @param {Date} fechaReserva - La nueva fecha de reserva.
     */
    setFechaReserva(fechaReserva) {
        this.fechaReserva = fechaReserva;
    }

    /**
     * Obtiene la fecha de devolución.
     * @returns {Date} La fecha de devolución.
     */
    getFechaDevolucion() {
        return this.fechaDevolucion;
    }

    /**
     * Establece la fecha de devolución.
     * @param {Date} fechaDevolucion - La nueva fecha de devolución.
     */
    setFechaDevolucion(fechaDevolucion) {
        this.fechaDevolucion = fechaDevolucion;
    }
}

// Exporta la clase Reserva
export default Reserva;
